README.md

```diff
@@ -97,59 +82,55 @@ To start using ANTLR Kotlin:
    }
    ```
 
-5. Disable the default ANTLR grammar generation task.  
-   This is set up by the Gradle ANTLR plugin, but it does not suit our needs.
+4. Register the ANTLR Kotlin grammar generation task.
 
    ```kotlin
-   tasks {
-     generateGrammarSource {
-       // The default task is set up considering a Java source set,
-       // which we might not have in a Kotlin project. 
-       // Using it is messier than simply registering a new task
-       enabled = false
-     }
-   }
-   ```
-
-6. Create the ANTLR Kotlin grammar generation task.
-
-   ```kotlin
-    val generateKotlinGrammarSource = tasks.register<AntlrTask>("generateKotlinGrammarSource") {
-        dependsOn("cleanGenerateKotlinGrammarSource")
-    
-        // ANTLR .g4 files are under {example-project}/antlr
-        setSource(layout.projectDirectory.dir("src/main/antlr"))
-    
-        val pkgName = "com.strumenta.antlrkotlin.parsers.generated"
-        arguments = listOf(
-            "-Dlanguage=Kotlin",    // We want to generate Kotlin sources
-            "-visitor",             // We want visitors alongside listeners
-            "-package", pkgName,    // We want the generated sources to have this package declared
-            "-encoding", "UTF-8",   // We want the generated sources to be encoded in UTF-8
-        )
-    
-        // Generated files are outputted inside build/
-        val outDir = "generatedAntlr/${pkgName.replace(".", "/")}"
-        outputDirectory = layout.buildDirectory.get().dir(outDir).asFile
+   val generateKotlinGrammarSource = tasks.register<AntlrKotlinTask>("generateKotlinGrammarSource") {
```

Comment:
As a follow-up, it would make sense for the gradle plugin to do this automatically, so that no custom configuration is needed

Reply:
It's tricky to setup tasks and task dependencies automatically. The reason I'm reluctant to do it is we might end up in the same situation as the Gradle `antlr` plugin: it works for some, and it breaks the project for others.

Reply:
I think we can make some logic controlled by a flag, to combine convenience for some with compatibility for others, but this is for sure a follow-up task. I created https://github.com/Strumenta/antlr-kotlin/issues/116 and I am closing this conversation, not to clutter the discussion